### PR TITLE
Add timing chart for libcudf build metrics report page

### PR DESCRIPTION
**Pull request by davidwendt**
Adds build time chart similar to ninjatracing output to the "Build Metrics Report" in a cudf build.

Other changes include
- showing times in minutes, seconds, or milliseconds as appropriate
- showing sizes in MB, KB, or bytes as appropriate
- adding a 4th color (white) for compile times less than 1s
- changed logic for identifying entries uniquely by hash instead of name (both in .ninja_log)

### CODE REVIEW

**Comment by davidwendt** [comment]
Example output of a nearly full build
https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-cuda-build/CUDA=11.5/7096/Build_20Metrics_20Report/

**Review by vuule — APPROVED** [review]
Looks good, just one nit: would it be possible/preferrable to represent the time in minutes as minutes:seconds (or maybe always use seconds)? Seeing 1.975 minutes is odd, at least to me.

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -85,6 +85,7 @@ if [ "$BUILD_LIBCUDF" == '1' ]; then
       gpuci_logger "Copying build metrics results"
       mkdir -p "$WORKSPACE/build-metrics"
       cp "$LIBCUDF_BUILD_DIR/ninja_log.html" "$WORKSPACE/build-metrics/BuildMetrics.html"
+      cp "$LIBCUDF_BUILD_DIR/ninja.log" "$WORKSPACE/build-metrics/ninja_log.html"
```
Comment (karthikeyann) [inline ci/cpu/build.sh:88]: Does this have to be `html` to be included in html publisher plugin?

Comment (davidwendt) [reply ci/cpu/build.sh:88]: I'm not sure. I do know this works.
I suppose it is worth trying.

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**